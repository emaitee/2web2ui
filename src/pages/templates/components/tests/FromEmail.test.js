import { shallow } from 'enzyme';
import React from 'react';

import FromEmailWrapper, { FromEmail } from '../FromEmail';

describe('From Email Typeahead Wrapper', () => {
  let wrapper;

  beforeEach(() => {
    const props = {
      input: {
        onChange: jest.fn()
      },
      meta: {},
      label: 'label'
    };

    wrapper = shallow(<FromEmailWrapper {...props} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render error', () => {
    wrapper.setProps({ meta: {
      active: false,
      touched: true,
      error: 'error message'
    }});
    expect(wrapper).toMatchSnapshot();
  });
});

describe('From Email Typeahead', () => {
  let wrapper;

  beforeEach(() => {
    const props = {
      domains: [
        { domain: 'test.com' },
        { domain: 'another.com' }
      ],
      onChange: jest.fn(),
      value: 'test@t'
    };

    wrapper = shallow(<FromEmail {...props} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should render list', () => {
    const args = {
      inputValue: 'test@t',
      isOpen: true,
      highlightedIndex: 0,
      getInputProps: jest.fn(),
      getItemProps: jest.fn((a) => a)
    };

    expect(wrapper).toHaveState('value', 'test@t');
    const result = wrapper.instance().typeaheadFn(args);
    expect(result).toMatchSnapshot();
  });

  it('should handle input value', () => {
    wrapper.instance().handleInputValueChange('new value');
    expect(wrapper).toHaveState('value', 'new value');
  });

  it('should handle state change', () => {
    const changes = {
      inputValue: 'new value'
    };
    const downshift = {
      highlightedIndex: null,
      setHighlightedIndex: jest.fn()
    };
    wrapper.instance().handleStateChange(changes, downshift);
    expect(wrapper.instance().props.onChange).toHaveBeenCalledWith('new value');
    expect(downshift.setHighlightedIndex).toHaveBeenCalledWith(0);
  });
});
