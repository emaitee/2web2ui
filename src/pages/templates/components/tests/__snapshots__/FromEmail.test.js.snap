// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`From Email Typeahead Wrapper should render 1`] = `
<FromEmail
  label="label"
  onChange={[Function]}
/>
`;

exports[`From Email Typeahead Wrapper should render error 1`] = `
<FromEmail
  error="error message"
  label="label"
  onChange={[Function]}
/>
`;

exports[`From Email Typeahead should render list 1`] = `
<div
  className={undefined}
>
  <TextField
    required={false}
    type="text"
  />
  <ActionList
    actions={
      Array [
        Object {
          "content": "test@test.com",
          "highlighted": true,
          "index": 0,
          "item": "test@test.com",
        },
      ]
    }
    className=""
  />
</div>
`;
